### DISCUSSION TITLE
Chaining the capture from the output of running a Hurl file as input for another Hurl file

### DISCUSSION THREAD

**Discussion by OctavianHome** [Orange-OpenSource/hurl#3455] [Q&A] [answered]
I want to use "json paths" from a Hurl script file (`file-001.hurl`) as input for a subsequent Hurl script file (`file-002.hurl`).
I know that I could use both Hurl script files (`file-001.hurl` and `file-002.hurl`) in the same Hurl file but this is not an option for me.
One potential solution (which is not quite convenient for me) would be to save the output of `file-001.hurl` to a `file-output-001.json` file, update that file to `file-output-001.env` file and then pass it as `--variables-file` to the `file-002.hurl`.
This is not convenient for me because:

- I have to process the `file-output-001.json` to `file-output-001.env` accepted by Hurl
- I have to pass the entire `file-output-001.json` content, regardless of what input (json path) I need

The ideal solution for me would be like bellow.
The `file-001.hurl` content:
```
GET https://jsonplaceholder.typicode.com/todos/1
HTTP 200
[Captures]
userId: jsonpath "$['userId']"
id: jsonpath "$['id']"
```
Then export `userid` and `id` using some cli flag (`--export-variables`) to an `output-001.env` file with this content:
```
userid=4
id=5
```
If `file-002.hurl` has this content:
```
GET https://users/{{userid}}/orders/{{id}}
HTTP 200
```
then I could pass `output-001.env` as input file for `file-002.hurl` like this:
```
hurl  --variables-file output-001.env file-002.hurl
```
Do you have any idea if something like this (or similar) can be done?

**Comment by jcamiel** [comment, marked as answer]
Hi @OctavianHome 

What can be done with Hurl current version is using `--json` : this way we have a structured output of the response and we can extract "easily" captured value.

With this file:

```hurl
GET https://jsonplaceholder.typicode.com/todos/1
HTTP 200
[Captures]
userId: jsonpath "$['userId']"
id: jsonpath "$['id']"
```

We can run it with `--json`:

```shell
$ hurl --json file1.hurl --output file1.json
$ # Get the userId value and the Id
$ USER_ID=$(cat file1.json | jq '.entries[0].captures[0].value')
$ ID=$(cat file1.json | jq '.entries[0].captures[1].value')
$ hurl --variable "userId=${USER_ID}" --variable "id=${ID}" file2.hurl
```

**Reply by OctavianHome** [reply]
Hi @jcamiel ,
I will try to see how is working like this.
Thanks for your suggestion.

**Comment by TimReddingPBX** [comment]
Hey, thought I'd share my approach as it works well with significantly less effort. 

I've found that concatenating the hurl files together using cat and piping them into the hurl cli works really well.
```
cat file1.hurl file2.hurl |  hurl   
```
